### ISSUE TITLE
Add Order tool to Configured Sites List (configuration)

### ISSUE THREAD

**Issue by vlc33** [Rayquaza01/authenticator#25]
## Feature Request

Hello, may you put an ordering tool so we can put order in the configured sites list?.
I do it mannually now crafting the json file, but maybe you can put some interface easing to do it.

In the mean time can you update and reduce the amount of text (notices and updates) before the configuration section? many new people could be afraid of so many letters, and even do not see the configuration interface downwards.

Thank You and congratulations again for this great job!
... and put you Bitcoin address so we can tip you!

**Comment by wazerstar** [comment]
Yes would be nice to just drag and and drop - up and down to make a perfect order, and while you are at it maybe make a category(spacer - where we can enter names for each section? - like games,social,etc..)

Think about it, I love your addon here!

**Comment by Rayquaza01** [comment]
@vlc33, @wazerstar 
Sorry for taking so long to get to this.
In the upcoming version, there will be "Up" and "Down" buttons next to each entry on the options page. Pressing up moves the entry up one position, and likewise for down.
Also, at the bottom of the options page, there is a "Sort Order" option. "Created" will sort by the order shown currently, "A-Z" will put them in alphabetical order, and "Z-A" will put them in reverse alphabetical order.
Could you test the pre-release to see if everything's working correctly before I publish it to AMO? https://github.com/Rayquaza01/authenticator/releases/tag/v1.3.3pre1

Thanks.

**Comment by wazerstar** [comment]
I dont like theres a scroll now after I only updated to this pre version

https://i.imgur.com/C86dj19.png

I dont like that you have to press a button up/down and really try to focus to move em, for me its seems like a real pain to click up/down and move mouse up and down and still see if you are hitting the correct move.

**Comment by vlc33** [comment]
I definetly vote for a 'drag to order interface' like in mobile apps, if possible.
I also dislike new scroll in runtime window

Thank You Very Much, this extensions rocks, and with so many articles around about 'sim cloning' of mobiles, TOPs as extensions is the answer isn't it? 

Have you think about introducing the password only once per sesion?

**Comment by Rayquaza01** [comment]
Hello,
Could you try the pre-release 2? https://github.com/Rayquaza01/authenticator/releases/tag/v1.3.3pre2
It should fix the scrollbar constantly showing. I also added the drag and drop sort tool. There is a "Drag and Drop Sort Tool" button above the list of sites. Pressing it activates drag and drop, and the "Save" button saves the order. Here's a video demo: https://www.youtube.com/watch?v=7TNR1WEkc4I.

As far as only accepting the password once per session, it wouldn't be able to work AFAIK. Right now, the extension stores the hash of the password and the encrypted secrets. You need to provide a password to decrypt the secrets. In order to have the password be remembered for the session, the password would either need to be stored somewhere in plain text, or the decrypted secrets would need to be stored somewhere in plain text. Both of those would defeat the purpose of encrypting in the first place.

Thanks.

**Comment by vlc33** [comment]
Fantastic, scrollbar has gone.

D&D is going in the right direction, I understand the drag mode in/out is for development purposes only.
As background, I just found that, just in case you did not already knew it
https://www.w3schools.com/HTML/html5_draganddrop.asp

On passwd, I understand, security has its own restrictions...

thx.

**Comment by wazerstar** [comment]
> 
> 
> Hello,
> Could you try the pre-release 2? https://github.com/Rayquaza01/authenticator/releases/tag/v1.3.3pre2
> It should fix the scrollbar constantly showing. I also added the drag and drop sort tool. There is a "Drag and Drop Sort Tool" button above the list of sites. Pressing it activates drag and drop, and the "Save" button saves the order. Here's a video demo: https://www.youtube.com/watch?v=7TNR1WEkc4I.
> 
> As far as only accepting the password once per session, it wouldn't be able to work AFAIK. Right now, the extension stores the hash of the password and the encrypted secrets. You need to provide a password to decrypt the secrets. In order to have the password be remembered for the session, the password would either need to be stored somewhere in plain text, or the decrypted secrets would need to be stored somewhere in plain text. Both of those would defeat the purpose of encrypting in the first place.
> 
> Thanks.

Thanks, for me its ok, still rough with the drag and drop but I understand its dev phase ;) works for me.

**Comment by wazerstar** [comment]
> 

> 
> Thanks.

Found a bug

Was adding  game auth with the site name and it bugged out with X chars.

More specific

escapefromtarkov.com as name

[](https://i.imgur.com/gyLBwG2.png)

**Comment by wazerstar** [comment]
Having more than 15 digits will make the sidebar scroll appear.
